server/server.go

```diff
@@ -362,3 +364,51 @@ func (s *Server) startGarbageCollection(ctx context.Context, frequency time.Dura
 	}()
 	return
 }
+
+// getUpdatedConnectorList retrieves a current list of connector objects available in
+// the storage. It gets the latest configuration of connectors that are dynamically
+// updated via the API.
+func (s *Server) getUpdatedConnectorList() error {
```

Comment:
So are you suggesting getting rid of the connectors map in the server struct all together and just making use of this function whenever we need the list of connectors?
Maybe returning a map here will make it easier to access a particular connection object.